db.go

```diff
@@ -934,6 +934,34 @@ func (db *DB) Begin() (*Tx, error) {
 	return &Tx{Tx: tx, DB: db}, nil
 }
 
+// Transaction begins a transaction as a block, if it errors it
+// will rollback and return an error, otherwise it will commit.
+func (db *DB) Transaction(fn func(tx *Tx) error) (err error) {
+	tx, err := db.Begin()
+
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			err = recoveryToError(r)
+		}
+
+		if err != nil {
+			if rollbackErr := tx.Rollback(); rollbackErr != nil {
+				err = combineErrors(err, rollbackErr)
+			}
+		}
+	}()
+
+	if err := fn(tx); err != nil {
```

Comment:
I have a concern that this `err` is shadowing the return value`err`, in which case the value of `err` in the `defer` will not be what you are expecting.